pci/src/bus.rs

```diff
@@ -484,3 +511,94 @@ fn parse_io_config_address(config_address: u32) -> (usize, usize, usize, usize)
         shift_and_mask(config_address, REGISTER_NUMBER_OFFSET, REGISTER_NUMBER_MASK),
     )
 }
+
+#[cfg(test)]
+mod tests {
+    use std::error::Error;
+    use std::result::Result;
+
+    use super::*;
+
+    #[derive(Debug)]
+    struct MocRelocDevice;
```

Comment:
please add a unit test where you specify devices with explicit addresses 1, 3, 5 and check that other devices are implicitley added at addresses 2 and 4